_posts/2021-05-07-running-multinode-modules-gcp.md

```diff
@@ -0,0 +1,174 @@
+  testing.
+* Teardown -  We use provision task to teardown the machines.
+
+Lets go through each step in detail.
+
+***Provision***
+
+Litmus allows the provisioning of a vm/container via a task.
+We use the provision module's [available tasks](https://github.com/puppetlabs/provision/blob/main/tasks) to spin up the test environment.
+When run it creates an litmus_inventory.yaml file that allows both [bolt](https://github.com/puppetlabs/bolt) and [serverspec](https://serverspec.org/) to communicate with that vm/container.
+
+How do we differentiate between different vm/containers? 
+Provisioning tasks in the provision module allow us to add arbitrary key/pair values to the bolt inventory file. 
+Using the bolt variables allows users to have multiple labels associated with a single machine.
+
+By running the provision bolt tasks manually or through a bolt plan we can label machines.
+
+![Showing role infm in litmus_inventory.yaml file]({% link /assets/2021-05-07-running-multinode-modules-gcp/inventory_role.png %}) 
+[Example_websphere_provision_plan](https://github.com/puppetlabs/puppetlabs-websphere_application_server/blob/main/plans/provision_machines.pp)
+
+***Setup provisioned systems***
+
+Setup for litmus in acceptance tests relied on 3 things.
+* Install agent - A bolt [task](https://github.com/puppetlabs/puppetlabs-puppet_agent/tree/main/tasks) that from a module which installs a puppet agent.
+* Install module - A rake task that uses the pdk to build the module under test, and install that on the target vms/containers.
+* spec/spec_helper_acceptance_local.rb - extra setup that may be required to test a module, Example [puppetlabs/puppetlabs-apache](https://github.com/puppetlabs/puppetlabs-apache/blob/main/spec/spec_helper_acceptance_local.rb)
+
+For integration testing we are using bolt plans and inventory variables where we can run puppet code / commands, against the specific systems, by labelling the systems.
+An example [puppetlabs-websphere_application_server](https://github.com/puppetlabs/puppetlabs-websphere_application_server/blob/main/plans/pe_server_setup.pp)
+
+![Example for provision plan]({% link /assets/2021-05-07-running-multinode-modules-gcp/provision_plan.png %})
+
+We can use spec/spec_helper_acceptance_local.rb for extra setup that may be required to test a module.
+An example [puppetlabs-websphere_application_server](https://github.com/puppetlabs/puppetlabs-websphere_application_server/blob/main/spec/spec_helper_acceptance_local.rb)
+
+***Run tests***
+
+We use rspec labelling and rake task to identify the tests to run on integeration environment. Lets go through the details.
+
+* Rspec labelling
+  How do we differentiate acceptance tests from integration tests?
+  Using rspec labeling we can label tests, e.g. the test is tagged as integration. Everything else can stay the same.
+
+  ![Example for tagging tests]({% link /assets/2021-05-07-running-multinode-modules-gcp/integration_tag.png %})
```

Comment:
Please change screenshots of code to textual code snippets for accessibility, e.g. that people can copy/paste from there.